crates/router/src/core/payments/routing.rs

```diff
@@ -233,16 +230,16 @@ pub async fn perform_static_routing_v1<F: Clone>(
     let key = ensure_algorithm_cached_v1(
         state,
         merchant_id,
-        algorithm_ref.timestamp,
         &algorithm_id,
         #[cfg(feature = "business_profile_routing")]
         payment_data.payment_intent.profile_id.clone(),
     )
     .await?;
     let cached_algorithm: Arc<CachedAlgorithm> = ROUTING_CACHE
-        .retrieve(&key)
+        .get_val(key.as_str())
```

Comment:
coalesce the `exists()` and `get_val()` calls into a single `get_val()` in the ensure_algorithm_cached function to prevent any errors that may arise if data gets invalidated between the `exists()` and `get_val()` calls.